### PR TITLE
[BUG] Windows compatibility with Makefile #308

### PR DESCRIPTION
**Pull request by Tokesh**
### Description
I added guidelines for Windows developers in the Developer Guide (to use WSL2 and how to use WSL)

### Issues Resolved

[#308 ]

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -68,6 +68,21 @@ In order to differentiate unit tests from integration tests, Go has a built-in m
    make cluster.stop cluster.clean
    ```
 
+#### Compatibility with Windows OS
+
+If you have a problem with running 'Make' follow this steps:
+
+1. Install WSL2 (compatibility layer for running Linux applications on Windows 10).
```
[the remark this answers is not in the document]
Comment (Tokesh) [reply DEVELOPER_GUIDE.md]: > I think you should just add a "Windows" section to prerequisites, with links/instructions to install/download WSL and ensure make works.
I think i got it, fixed that in new commit. I tried to make it shorter like you said in Issue discussion

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -32,6 +33,15 @@ OpenSearch Go Client builds using [Go](https://golang.org/doc/install) 1.11 at a
 
 [Docker](https://docs.docker.com/install/) is required for building some OpenSearch artifacts and executing integration tests.
 
+#### Windows
+
+[WSL2](https://learn.microsoft.com/en-us/windows/wsl/install) is compatibility layer for running Linux applications on Windows.
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:38]: Missing "the", but let's say "To build the project on Windows, use WSL2, the compatibility ...".

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -32,6 +33,15 @@ OpenSearch Go Client builds using [Go](https://golang.org/doc/install) 1.11 at a
...
+
+Make ensure that you install 'Make' on WSL2.
```
Comment (dblock) [inline DEVELOPER_GUIDE.md:40]: Install `make`.

**Review by dblock — CHANGES_REQUESTED** [review]
The first phrase needs some english fixing :) Thanks for hanging in here with me!

File: DEVELOPER_GUIDE.md (hunk 2 above)
Comment (Tokesh) [reply DEVELOPER_GUIDE.md:38]: > The first phrase needs some english fixing :) Thanks for hanging in here with me!
I appreciate that you are patient with me and I am glad to work together with you!

**Review by dblock — APPROVED** [review]

**Pull request merged**